### ISSUE TITLE
Interface produces no-undef errors

### ISSUE THREAD

**Issue by nickmccurdy** [eslint/typescript-eslint-parser#437]
**What version of TypeScript are you using?**
2.6.2

**What version of `typescript-eslint-parser` are you using?**
12.0.0

**What code were you trying to parse?**

```ts
interface Runnable {
  run (): Result
  toString (): string
}
```

**What did you expect to happen?**
Pass with no errors.

**What happened?**
```
> eslint **/*.{j,t}s


/Users/nick/Repos/purespec/matchers/Runnable.ts
  1:11  error  'Runnable' is not defined  no-undef
  2:3   error  'run' is not defined       no-undef
  2:11  error  'Result' is not defined    no-undef

✖ 3 problems (3 errors, 0 warnings)
```

**Comment by sanpoChew** [comment]
@nickmccurdy did you find a solution to this? thanks

**Comment by EdGraVill** [comment]
This is resolved in #75

**Comment by Bessonov** [comment]
Unfortunately, #75 isn't resolution.

**Comment by jacob-israel-turner** [comment]
Has there been any effort on this?  If not, I'd be willing to take a stab - although I have very little experience writing linting rules.  Is there somewhere I could look for a good first step?

**Comment by amcdnl** [comment]
Any updates on this? With the new Babel7 TypeScript support this becomes more interesting.

**Comment by madou** [comment]
Alternatively you could disable the rule and just use ts compiler settings to error on unused variables?

https://www.typescriptlang.org/docs/handbook/compiler-options.html

`--noUnusedLocals` `--noUnusedParameters`
